File: 56MergeIntervals/mergeIntervals.py
from typing import List

class Solution:
  def merge(self, intervals: List[List[int]]) -> List[List[int]]:
    intervals.sort(key = lambda p: p[0])
    ans = []
    for it in intervals:
      if not ans or ans[-1][1] < it[0]:
        ans.append(it)
      elif ans[-1][1] <= it[1]:
        ans[-1][1] = it[1]
    return ans

File: 56MergeIntervals/mergeIntervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (int[] a, int[] b) -> {
            return a[0] - b[0];
        });
        List<int[]> ans = new ArrayList<>();
        for (int[] x : intervals) {
            int n = ans.size();
            if (ans.isEmpty() || ans.get(n - 1)[1] < x[0]) {
                ans.add(x);
            } else if (ans.get(n - 1)[1] <= x[1]) {
                ans.get(n - 1)[1] = x[1];
            }
        }
        return ans.toArray(new int[ans.size()][]);
    }
}
